web/src/components/storage/VolumeFields.jsx

```diff
@@ -461,271 +444,56 @@ const SIZE_OPTION_LABELS = Object.freeze({
+ * @property {(v: object) => void} onChange - callback for notifying input changes
+ *
+ * @param {SizeOptionsFieldProps} props
  */
-const SizeOptions = ({ errors, formData, volume, onChange }) => {
+const SizeOptionsField = ({ volume, formData, isDisabled = false, errors = {}, onChange }) => {
   const { sizeMethod } = formData;
-  const sizeWidgetProps = { errors, formData, volume, onChange };
+  const sizeWidgetProps = { errors, formData, volume, isDisabled, onChange };
 
   /** @type {string[]} */
   const sizeOptions = [SIZE_METHODS.MANUAL, SIZE_METHODS.RANGE];
 
   if (volume.outline.supportAutoSize) sizeOptions.push(SIZE_METHODS.AUTO);
 
   return (
-    <div>
-      <div className="split radio-group">
-        { sizeOptions.map((value) => {
-          const isSelected = sizeMethod === value;
-
-          return (
-            <Radio
-              id={value}
-              key={`size-${value}`}
-              // eslint-disable-next-line agama-i18n/string-literals
-              label={_(SIZE_OPTION_LABELS[value] || value)}
-              value={value}
-              name="size-option"
-              className={isSelected && "selected"}
-              isChecked={isSelected}
-              onChange={() => onChange({ sizeMethod: value })}
-            />
-          );
-        })}
-      </div>
-
-      <div aria-live="polite" className="highlighted-live-region">
-        <If condition={sizeMethod === SIZE_METHODS.AUTO} then={<SizeAuto { ...sizeWidgetProps } />} />
-        <If condition={sizeMethod === SIZE_METHODS.RANGE} then={<SizeRange {...sizeWidgetProps } />} />
-        <If condition={sizeMethod === SIZE_METHODS.MANUAL} then={<SizeManual { ...sizeWidgetProps } />} />
+    <FormGroup role="radiogroup" fieldId="size" label={_("Size")} isRequired>
+      <div>
+        <div className="split radio-group">
+          {sizeOptions.map((value) => {
+            const isSelected = sizeMethod === value;
+
+            return (
+              <Radio
+                id={value}
+                key={`size-${value}`}
+                // eslint-disable-next-line agama-i18n/string-literals
+                label={_(SIZE_OPTION_LABELS[value] || value)}
+                value={value}
+                name="size-option"
+                className={isSelected && "selected"}
+                isChecked={isSelected}
+                onChange={() => onChange({ sizeMethod: value })}
+                isDisabled={isDisabled}
+              />
+            );
+          })}
+        </div>
+
+        <div aria-live="polite" className="highlighted-live-region">
+          <If condition={sizeMethod === SIZE_METHODS.AUTO} then={<SizeAuto {...sizeWidgetProps} />} />
+          <If condition={sizeMethod === SIZE_METHODS.RANGE} then={<SizeRange {...sizeWidgetProps} />} />
+          <If condition={sizeMethod === SIZE_METHODS.MANUAL} then={<SizeManual {...sizeWidgetProps} />} />
+        </div>
```

Comment:
Intentionally I marked tests for this as "todo" because I plan to change this approach a bit in order to improve the form. Out of scope of this PR.